File: solutions/javascript/space-age/1/space-age.js
const getEarthYears = (planet, seconds) => {
  const earthYears = seconds / 31557600;

  switch (planet.toLowerCase()) {
    case 'mercury':
      return earthYears / 0.2408467;
    case 'venus':
      return earthYears / 0.61519726;
    case 'earth':
      return earthYears;
    case 'mars':
      return earthYears / 1.8808158;
    case 'jupiter':
      return earthYears / 11.862615;
    case 'saturn':
      return earthYears / 29.447498;
    case 'uranus':
      return earthYears / 84.016846;
    case 'neptune':
      return earthYears / 164.79132;
    default:
      throw new Error('Invalid planet.');
  }
};

export const age = (planet, seconds) => {
  return +getEarthYears(planet, seconds).toFixed(2);
};

File: solutions/c/space-age/1/space_age.c
#include "space_age.h"

float age(planet_t planet, int64_t seconds)
{
    float earth = (float)seconds / 31557600.0;

    if (planet == MERCURY)
        return earth / 0.2408467;

    if (planet == VENUS)
        return earth / 0.61519726;

    if (planet == EARTH)
        return earth;

    if (planet == MARS)
        return earth / 1.8808158;

    if (planet == JUPITER)
        return earth / 11.862615;

    if (planet == SATURN)
        return earth / 29.447498;

    if (planet == URANUS)
        return earth / 84.016846;

    if (planet == NEPTUNE)
        return earth / 164.79132;

    return -1.0;
}
